tests/diff_context_test.py

```diff
@@ -161,31 +159,3 @@ async def test_diff_context_pr(temp_testbed, git_history, mock_session_context):
     assert diff_context.name.startswith("Merge-base Branch master:")
     assert diff_context.name.endswith(": commit2")  # NOT the latest
     assert diff_context.files == [abs_path]
-
-
-@pytest.mark.asyncio
-async def test_diff_context_end_to_end(
-    temp_testbed, git_history, mock_call_llm_api, mock_setup_api_key
-):
```

Comment:
why is this removed?

Reply:
It would've been an annoying test to port over, and I didn't think it was adding any coverage; we don't really have any other 'e2e' tests like this one (just client tests, but because the Session keeps the test SessionContext they work pretty easily), just unit tests, and in order to add them I'm going to add a marker to not automock session context, add a fixture to mock llm_api_handler, and change a few things. I'm actually thinking I'm going to change  it so we don't mock SessionContext though, since I realized it's a bit of a relic making tests harder though. I probably won't add this test back though, since like I said it doesn't really add any coverage.

Reply:
Re-added the test!